sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -431,6 +461,9 @@ export class EventHubReceiver extends LinkEntity {
           this.address
         );
       } else {
+        if (this._aborter) {
+          this._aborter.removeEventListener("abort", this._onAbort);
+        }
```

Comment:
Instead of removing the listener in 2 places inside detached(), we can add an `else` for the `if (shouldReopen)` and remove it just there